packages/js/plugins/http/src/types.ts

```diff
@@ -11,6 +11,8 @@ export class UrlParam {
   value: string;
 }
 
+// even though in schema enum is used, this will be converted to string
+// so this is a reason to define ResponseType as string here
```

Comment:
Where is this conversion happening? Enums coming from Web3APIs will be integers.

Reply:
Changed it to `enum`, I didn't refactor tests as I should and that mislead me to the wrong conclusion. Now everything should be good.

Reply:
For the tests when you're sending a query into WASM, you can use a string or a number. But that string gets turned into a number immediately in the deserialization code, and it remains a number from there on out.